test/format/backup.c

```diff
@@ -348,6 +372,104 @@ copy_file(WT_SESSION *session, const char *name)
  */
 #define HOME_BACKUP_INIT_CMD "rm -rf %s/BACKUP %s/BACKUP.copy && mkdir %s/BACKUP %s/BACKUP.copy"
 
+/*
+ * restore_backup_info --
+ *     If it exists, restore the backup information. Return 0 on success.
+ */
+static int
+restore_backup_info(WT_SESSION *session, ACTIVE_FILES *active)
+{
+    FILE *fp;
+    WT_CURSOR *cursor;
+    WT_DECL_RET;
+    size_t len;
+    uint64_t id;
+    uint32_t i;
+    char buf[512], *path;
+
+    testutil_assert(g.c_backup_incr_flag == INCREMENTAL_BLOCK);
+    len = strlen(g.home) + strlen(BACKUP_INFO_FILE) + 2;
+    path = dmalloc(len);
+    testutil_check(__wt_snprintf(path, len, "%s/%s", g.home, BACKUP_INFO_FILE));
+    if ((fp = fopen(path, "r")) == NULL && errno != ENOENT)
+        testutil_die(errno, "restore_backup_info fopen: %s", path);
+    free(path);
+    if (errno == ENOENT)
+        return (1);
+    ret = fscanf(fp, "%" SCNu64 "\n", &id);
+    if (ret != 1)
+        testutil_die(EINVAL, "restore_backup_info ID");
+    /*
+     * Try to open the backup cursor. We may get ENOENT if the source ID we wrote to the program
+     * file was not yet checkpointed. Sometimes it will, sometimes it won't. If we don't find it
+     * then return non-zero so that we skip incremental restart testing.
+     *
+     * NOTE: This call to open a backup cursor to check the source id uses a made up 'this_id' that
+     * tries to generate one that cannot possibly be in use. This call can/should be changed if the
+     * API ever allows us to open a cursor with a source id that does not require a this id.
+     */
+    testutil_check(__wt_snprintf(buf, sizeof(buf),
+      "incremental=(enabled,src_id=%" PRIu64 ",this_id=%" PRIu64 ")", id, id / 2));
+    while ((ret = session->open_cursor(session, "backup:", NULL, buf, &cursor)) == EBUSY)
+        __wt_yield();
+    if (ret != 0) {
+        if (ret == ENOENT)
+            return (1);
+        testutil_die(ret, "session.open_cursor: backup");
+    }
+    testutil_check(cursor->close(cursor));
+
+    active_files_init(active);
+    ret = fscanf(fp, "%" SCNu32 "\n", &active->count);
+    if (ret != 1)
+        testutil_die(EINVAL, "restore_backup_info active count");
+
+    /* Set global id after error paths. */
+    g.backup_id = id + 1;
+    active->names = drealloc(active->names, sizeof(char *) * active->count);
+    for (i = 0; i < active->count; ++i) {
+        memset(buf, 512, 0);
+        fscanf(fp, "%s\n", buf);
```

Comment:
Probably want to limit the string size `%511s` ?  and check the return value of `fscanf`